File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void) {
    string text = get_string("Text: ");
    int sentences = 0;
    int letters = 0;
    int words = 1;

    for (int i = 0; text[i] != '\0'; i++) {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?') { //COUNT NUMBER OF SENTENCES WE HAVE
            sentences ++;
        }

        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122)) { //COUNT NUMBER OF LETTERS
            letters ++;
        }

        if (text[i] == ' ') { //COUNTS NUMBER OF WORDS
            words ++;
        }
    }

    /*printf ("Sentences: %i\n", sentences);
    printf ("Letters: %i\n", letters);
    printf ("Words: %i\n", words);*/

    float lettersperwords = letters*(100.0/words);
    float sentencesperwords = sentences*(100.0/words);

    /*printf ("letters per words: %f\n", lettersperwords);
    printf ("sentence per words: %f\n", sentencesperwords);*/

    float gradelvl = round((0.0588 * lettersperwords) - (0.296 * sentencesperwords) - 15.8);

    if (gradelvl < 1) {
        printf ("Before Grade 1\n");
    } else if (gradelvl >= 16) {
        printf ("Grade 16+\n");
    } else {
        printf ("Grade %.0f\n", gradelvl);
    }
}

File: pset6/readability/readability.py
txt = input("Text: ")

sentences = 0
letters = 0
words = 1

for lttr in txt:
    if (lttr == "!" or lttr == "." or lttr == "?"):
        sentences += 1

    if (lttr.isalpha()):
        letters += 1

    if (lttr == " "):
        words += 1

lettersperwords = letters*(100/words)
sentencesperwords = sentences*(100/words);

gradelvl = round((0.0588 * lettersperwords) - (0.296 * sentencesperwords) - 15.8)

if (gradelvl < 1):
    print("Before Grade 1");
elif (gradelvl >= 16):
    print("Grade 16+");
else:
    print("Grade " + str(gradelvl));